src/Language/Mimsa/Interpreter/Interpret.hs

```diff
@@ -165,9 +161,7 @@ interpretWithScope interpretExpr =
       predExpr <- interpretWithScope pred'
       interpretWithScope (MyIf ann predExpr true false)
     (MyData _ _ expr) -> interpretWithScope expr
-    (MyConstructor ann a) -> pure (MyConstructor ann a)
-    (MyConsApp ann fn val) ->
-      MyConsApp ann fn <$> interpretWithScope val
     (MyCaseMatch _ expr' matches catchAll) -> do
       expr'' <- interpretWithScope expr'
       patternMatch expr'' matches catchAll >>= interpretWithScope
+    expr -> bindExpr interpretWithScope expr
```

Comment:
This catch-all just does the sensible thing - run `interpretWithScope` on any nested `Expr`s.